File: quicksort.js
function quicksort(array) {
    if (array.length < 2) return array    

    
    let middle = Math.floor((array.length - 1) / 2)
    let pivot = array[middle]    
    let lowers = array.filter((element) => element < pivot) 
    let equals = array.filter((element) => element == pivot)
    let greaters = array.filter((element) => element > pivot)

    return quicksort(lowers).concat(equals.concat(quicksort(greaters)))
}

console.log(quicksort([]))
console.log(quicksort([20]))
console.log(quicksort([33, 15 ,10]))
console.log(quicksort([33, 15, 10, 20, 25, 12, 13, 22]))

File: quicksort.py
from math import floor

array = [30, 15, 5, 10]
lowers = []
greaters = []
equals = []

def quicksort(array):

    if len(array) < 2:
        return array 
    
    middle = floor(len(array) - 1 / 2)
    pivot = array[middle]
    
    lowers = list(filter(lambda x: x < pivot, array))
        
    equals = list(filter(lambda x: x == pivot, array))
        
    greaters = list(filter(lambda x: x > pivot, array))

    return quicksort(lowers) + equals + quicksort(greaters)


print(quicksort(array))

